json/src/value.rs

```diff
@@ -473,6 +559,99 @@ impl de::Deserialize for Value {
     }
 }
 
+/// Represents type of a JSON value
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+pub enum ValueType {
```

Comment:
I agree. The library is still unstable, so we can wait. I hope time is not concern to anyone else.

I think I'm going to leave this as is for now until discriminant is stabilized. However, I'm not sure whether discriminant will allow to do exactly what I mean. I will check that out.